tests/expressions/tests.py

```diff
@@ -591,6 +591,14 @@ def test_pickle_expression(self):
         expr.convert_value  # populate cached property
         self.assertEqual(pickle.loads(pickle.dumps(expr)), expr)
 
+    def test_incorrect_field_expression(self):
+        with self.assertRaisesMessage(FieldError, "Cannot resolve keyword 'nope' into field."):
+            list(Employee.objects.filter(firstname=F('nope')))
+
+    def test_incorrect_field_expression_in_join(self):
+        with self.assertRaisesMessage(FieldError, "Cannot resolve keyword 'nope' into field."):
+            list(Company.objects.filter(ceo__firstname=F('point_of_contact__nope')))
```

Comment:
Running `test_incorrect_field_expression_in_join()` without a fix, doesn't behave like it's described in the ticket on the PostgreSQL database. It raises: 

_"django.db.utils.ProgrammingError: operator does not exist: character varying = integer"_

IMO using `ceo__pk` instead of `ceo_firstname` should fix this issue.